### PR TITLE
HPCC-29282 Show Dali size and load progress

### PR DESCRIPTION
**Pull request by jakesmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: system/jlib/jfile.cpp [hunk 1]
```diff
@@ -4441,6 +4441,76 @@ IFileIOStream * createBufferedIOStreamFromFile(const char *fileNameWithPath, IFO
...
+        }
+        virtual size32_t write(size32_t len, const void * data) override
+        {
+            throwUnexpected();
+        }
+        // implements IFileIOStream
+        virtual void seek(offset_t pos, IFSmode origin) override
+        {
+            return iFileIOStream->seek(pos, origin);
+
```
Comment (ghalliday) [inline system/jlib/jfile.cpp:4496]: trivial: newline

File: system/jlib/jfile.cpp [hunk 2]
```diff
@@ -4441,6 +4441,76 @@ IFileIOStream * createBufferedIOStreamFromFile(const char *fileNameWithPath, IFO
...
+        {
+            offset_t pos = iFileIOStream->tell();
+            if (pos == totalSize)
+                log(100.0);
+        }
+        // implements ISimpleReadStream
+        virtual size32_t read(size32_t max_len, void * data) override
+        {
+            cycle_t nowCycles = get_cycles_now();
+            if (nowCycles >= nextLogCycles)
```
Comment (ghalliday) [inline system/jlib/jfile.cpp:4474]: Trivial: This could theoretically wrap - although would probably need an uptime of 100 years...

**Review by ghalliday — APPROVED** [review]
@jakesmith looks good.  One trivial formatting issue, but would merge.
Might be worth looking at https://github.com/hpcc-systems/HPCC-Platform/pull/17206 and using that?

File: system/jlib/jfile.cpp (hunk 1 above)
Comment (jakesmith) [reply system/jlib/jfile.cpp]: removed, squashed, pushed.

File: system/jlib/jfile.cpp (hunk 2 above)
Comment (jakesmith) [reply system/jlib/jfile.cpp:4474]: yes, I did realize, but didn't think worth worrying about too much, and there are many other places that would similarly fall foul if this was so, including all uses of CCycleTimer::elapsed* functions.

**Comment by jakesmith** [comment]
> @jakesmith looks good. One trivial formatting issue, but would merge. Might be worth looking at #17206 and using that?

yes, probably. I can use once you merge that.

**Comment by ghalliday** [comment]
@jakesmith I have merged that other PR.  Do you want me to merge this as-is, or experiment with using that class?

**Comment by jakesmith** [comment]
> @jakesmith I have merged that other PR. Do you want me to merge this as-is, or experiment with using that class?

I'll incorporate the new helper class in this PR.

File: system/jlib/jfile.cpp (hunk 2 above)
Comment (jakesmith) [reply system/jlib/jfile.cpp]: > > @jakesmith I have merged that other PR. Do you want me to merge this as-is, or experiment with using that class?
> 
> I'll incorporate the new helper class in this PR.

@ghalliday - have now incorporated PeriodicTimer

**Comment by jakesmith** [comment]
@ghalliday - it needed rebasing to pick up PeriodicTimer

**Review by ghalliday — COMMENTED** [review]
@jakesmith please squash.  Did you test the new code?

**Comment by jakesmith** [comment]
> @jakesmith please squash. Did you test the new code?

@ghalliday - yes I tested. Now squashed.

**Pull request merged**